lbz/authz.py

```diff
@@ -148,12 +111,13 @@ def _check_allow(self):
             if d_domain := self.allow.get(self.resource):
                 if self._allow_if_allow_all(d_domain):
                     return
-                elif resource_to_check := d_domain.get(self.action):
+                elif resource_to_check := d_domain.get(self.permission):
                     self.outcome = ALLOW
                     self.allowed_resource = resource_to_check.get("allow")
                     self.denied_resource = resource_to_check.get("deny")
 
```

Comment:
Could you please rename `check_allow` to `check_allow_and_set_resources` ? IMHO this method needs to be redesigned, but for now I would at least make it clear that this function does not only one thing (which might be helpfull for debugging).